app/development/scripts/clear_app_data.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+# assert ANDROID_HOME is set
+if [ -z "$ANDROID_SDK_HOME" ]; then
+  echo "ANDROID_SDK_HOME is not set"
+  exit 1
+fi
+
+SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
+DEVELOPMENT_DIR=$SCRIPT_DIR/..
+ROOT_PROJECT_DIR=$SCRIPT_DIR/../../..
+
+EMULATOR_DEVICE_NAME=$($ANDROID_SDK_HOME/platform-tools/adb devices | grep emulator | cut -f1)
```

Comment:
this limits execution to emulators. is this intentional? why not allow physical test devices?

Reply:
The reason I did this was to avoid the person running the tests accidentially targeting their personal phone (like me :laughing:) when running these tests, since it will do some destructive things like uninstalling packages, clearing backups, wiping seedvault, etc.

I can't test on a physical device unfortunately, but if we could think of away around the above could avoid this I'm sure?

Reply:
It does, but for instance this script in particular could be run against a personal device and clear the Seedvault data. Maybe just need to enforce only one device is connected?

Reply:
@chirayudesai is there a way to check for a userdebug build device? Maybe that would be more useful than single device?

Reply:
`getprop ro.build.type` - user, userdebug, or eng

Also, `ro.build.tags`. `test-keys` for dev, `release-keys` for prod (typically)

Ref: https://dumps.tadiphone.dev/dumps/google/lynx/-/blob/lynx-user-13-TQ3A.230605.012-10204971-release-keys/system/system/build.prop?ref_type=heads